### DISCUSSION TITLE
Cam qui fonctionne plus

### DISCUSSION THREAD

**Discussion by Elcidz** [Guilouz/Klipper-Flsun-Speeder-Pad#110] [Q&A]
Salut Guillouz j arrive plus à connecter ma Cam C922 pro j ai tout réinstaller rien a faire tu a une idée d ou sa peut venir sa marcher bien avant les mises à jour proposées j ai désinstallé Crowsnest et réinstaller rien à faire merci à toi

**Comment by Guilouz** [comment]
Tente ça en SSH : 

cd ~/crowsnest
make clean
make buildclean
cd bin
./build.sh -r
cd ~/crowsnest
make

**Reply by Elcidz** [reply]
Merci Cyril

Je test je te dit si sa fonctionne

**Reply by Elcidz** [reply]
dit moi je desinstall avant ou pas et je fais quoi ? aprés crowsnest installé j ai balancé tes commande mais toujours pareil
merci

pi@speeder-pad:~$ cd ~/crowsnest
pi@speeder-pad:~/crowsnest$ make clean
Removed installer config file ...
pi@speeder-pad:~/crowsnest$ make buildclean
bash -c 'bin/build.sh --clean'

Running 'make clean' in /home/pi/crowsnest/bin/ustreamer ...
make[1]: Entering directory '/home/pi/crowsnest/bin/ustreamer'
rm -rf pkg/arch/pkg pkg/arch/src pkg/arch/v*.tar.gz pkg/arch/ustreamer-*.pkg.tar.{xz,zst}
rm -f ustreamer ustreamer-dump *.so
make -C src clean
make[2]: Entering directory '/home/pi/crowsnest/bin/ustreamer/src'
rm -rf ustreamer.bin ustreamer-dump.bin build
make[2]: Leaving directory '/home/pi/crowsnest/bin/ustreamer/src'
make -C python clean
make[2]: Entering directory '/home/pi/crowsnest/bin/ustreamer/python'
rm -rf build
make[2]: Leaving directory '/home/pi/crowsnest/bin/ustreamer/python'
make -C janus clean
make[2]: Entering directory '/home/pi/crowsnest/bin/ustreamer/janus'
rm -rf libjanus_ustreamer.so build
make[2]: Leaving directory '/home/pi/crowsnest/bin/ustreamer/janus'
make[1]: Leaving directory '/home/pi/crowsnest/bin/ustreamer'

Running 'make clean' in /home/pi/crowsnest/bin/camera-streamer ...
make: *** [Makefile:26: buildclean] Error 1
pi@speeder-pad:~/crowsnest$ cd bin
pi@speeder-pad:~/crowsnest/bin$ ./build.sh -r
Deleting '/home/pi/crowsnest/bin/ustreamer' ...
'/home/pi/crowsnest/bin/camera-streamer' does not exist! Delete skipped ...
Cloning into '/home/pi/crowsnest/bin/ustreamer'...
remote: Enumerating objects: 174, done.
remote: Counting objects: 100% (174/174), done.
remote: Compressing objects: 100% (147/147), done.
remote: Total 174 (delta 61), reused 67 (delta 14), pack-reused 0
Receiving objects: 100% (174/174), 173.43 KiB | 1.51 MiB/s, done.
Resolving deltas: 100% (61/61), done.
WARN: Cloning camera-streamer skipped! Device is not supported!pi@speeder-pad:~/crowsnest/bin$ cd ~/crowsnest
pi@speeder-pad:~/crowsnest$ make
crowsnest - A webcam Service for multiple Cams and Stream Services.
Usage:

install              Install crowsnest (needs leading sudo)
uninstall            Uninstall crowsnest
build                Compile backends / streamer
buildclean           Clean backends / streamer (for rebuilding)
clean                Clean .config
config               Configure crowsnest installer
help                 Shows this help
update               Update crowsnest (fetches and pulls repository changes)
report               Generate report.txt

**Comment by Guilouz** [comment]
C’est donc que ça ne vient pas de Crowsnest donc peut-être de la config de ta cam. Elle est détectée en ssh?

**Reply by Elcidz** [reply]
Je vois sa comment tout marchait bien avant la elle s allume plus j  ai test sur pc pas de pb j ai tester sur un autre port du pad sa marche pas non plus j ai réinstaller ta configuration 4 fois sa change rien là je vois pas. comment je test en ssh ? Dsl Linux s est pas mon truc après te prend pas la tête s est pas vital je pense que tu as beaucoup questions en cours je peux attendre merci encore

**Comment by Guilouz** [comment]
Avec la commande : ls /dev/v4l/by-id/*

**Comment by Elcidz** [comment]
pi@speeder-pad:~$ ls /dev/v4l/by-id/*
/dev/v4l/by-id/usb-046d_C922_Pro_Stream_Webcam_7891316F-video-index0
j ai sa donc je pense que s est bon enfin je pense

**Reply by Guilouz** [reply]
Et ton fichier de conf crowsnest il ressemble à quoi?

**Reply by Elcidz** [reply]
Salut Cyril
J ai sa dans le fichier j ai rien modifié

[crowsnest]
log_path: /home/pi/printer_data/logs/crowsnest.log
log_level: verbose                      # Valid Options are quiet/verbose/debug
delete_log: false                       # Deletes log on every restart, if set to true
no_proxy: false

[cam 1]
mode: ustreamer                         # ustreamer - Provides mjpg and snapshots. (All devices)
                                        # camera-streamer - Provides webrtc, mjpg and snapshots. (rpi + Raspi OS based only)
enable_rtsp: false                      # If camera-streamer is used, this enables also usage of an rtsp server
rtsp_port: 8554                         # Set different ports for each device!
port: 8080                              # HTTP/MJPG Stream/Snapshot Port
device: /dev/video0                     # See Log for available ...
resolution: 640x480                     # widthxheight format
max_fps: 15                             # If Hardware Supports this it will be forced, otherwise ignored/coerced.
#custom_flags:                          # You can run the Stream Services with custom flags.
#v4l2ctl:                               # Add v4l2-ctl parameters to setup your camera, see Log what your cam is capable of.

Merci

**Reply by Elcidz** [reply]
Salut Cyril,
Bon je viens de trouver une solution et fait je suis obliger de redémarrer le service Crowsnest a chaque fois que je redémarre le pad est la sa fonctionne j ai regardé dans le log file 
[05/30/23 18:20:38] crowsnest: Try to start configured Cams / Services...
[05/30/23 18:20:41] crowsnest: INFO: Configuration of Section [cam 1] looks good. Continue ...
[05/30/23 18:20:41] crowsnest: V4L2 Control: No parameters set for [cam 1]. Skipped.
[05/30/23 18:20:44] crowsnest: Starting ustreamer with Device /dev/video0 ...
[05/30/23 18:20:44] crowsnest: ERROR: Start of ustreamer [cam 1] failed!
[05/30/23 18:20:45] crowsnest:  ... Done!

J ai une erreur
Après le redémarrage du service sa fonctionne
Merci a toi
